Список будущих поездок и подробности будущей поездки #71

valerachuk opened the pull request.

valerachuk committed:
#71 display & delete waypoints

valerachuk committed:
#71 create & edit waypoints

valerachuk committed:
#71 add translations

move route params to template props in waypoint components
optimize data load logic

valerachuk committed:
#71 add trip-add form

add google place image api

valerachuk committed:
#71 fix checkboxes

valerachuk committed:
#71 add waypoint files

horba commented on front/src/components/mmt-waypoint-card/mmt-waypoint-card.component.html:
```diff
@@ -0,0 +1,95 @@
+    <v-card class="card-wrapper" color="primary lighten-7" tile flat>
+      <div class="delete-btn" @click="deleteWaypoint">
+        X
+      </div>
```
you didn't need to create workaround - just use v-button with delete icon

horba replied:
```
<v-btn icon>
    <v-icon>mdi-delete</v-icon>
</v-btn>
```

horba commented on front/src/components/mmt-waypoint-card/mmt-waypoint-card.component.html:
```diff
@@ -0,0 +1,95 @@
+          </div>
+        </div>
+        <div class="arrival" v-else>
+          <div class="date">{{$t('tripHistory.arrivalDate')}} {{waypoint.arrivalDate | date('d MMM y, HH:mm')}}</div>
```
why arrival date twice?

valerachuk replied:
![image](https://user-images.githubusercontent.com/38440869/85748641-e22a3c80-b710-11ea-8f25-36806c497cab.png)
first one is just translate

horba commented on front/src/components/mmt-waypoint-card/mmt-waypoint-card.component.html:
```diff
@@ -0,0 +1,95 @@
+          <div class="date">{{$t('tripHistory.arrivalDate')}} {{waypoint.arrivalDate | date('d MMM y, HH:mm')}}</div>
+          <div class="city">
+            <span class="subtitle">
+              {{$t('tripHistory.goingTo')}}
```
what is goingTo?

valerachuk replied:
translate - "Собираюсь в"

horba commented on front/src/components/mmt-waypoint-card/mmt-waypoint-card.component.html:
```diff
@@ -0,0 +1,95 @@
+        <div v-else>
+        </div>
+        <div class="details">
+          <v-checkbox hide-details readonly color="#2962FF" :input-value="waypoint.isDetails" @click.self="updateDetailsState"/>
```
if it is not primary, secondary etc - better to do via css

horba commented on front/src/components/mmt-waypoint-card/mmt-waypoint-card.component.js:
```diff
@@ -0,0 +1,38 @@
+    firstCard: {
+      type: Boolean,
+      default: false
+    },
```
it is not what i expect from card component. card should render only card, no other things and it don't need to know its order
it seems just it is not only card - so, separate non card part to other component

valerachuk replied:
But first card should not contain arrival date, last card - departure date

horba commented on front/src/router/index.js:
```diff
@@ -166,11 +167,45 @@ const routes = [
+        props: true
+      },
+      {
+        path: 'history/future/:id/route/:wpId/edit',
```
wp - not shortcuts - it is unclear that it is way point at all

horba commented on front/src/store/modules/waypoints.store.js:
```diff
@@ -0,0 +1,111 @@
+      commit('CLEAR_WAYPOINTS');
+      return api.post('/waypoints', wp)
+        .then(() => {
+          dispatch('loadWaypoints', [state.tripId, true]);
```
please, not shortcuts

horba commented on front/src/styles/variables.css:
```diff
@@ -22,4 +22,8 @@
   --user-menu-disabled-color: #B6B6B6;
   --mmt-avatar-color: #195292;
+  --history-route-transport-icon-color: #00C853;
+  --history-route-details-color: #195292;
```
it seems we already have color, migth we need more common name? it seems as mono pallet

horba commented on front/src/views/index.js:
```diff
@@ -26,3 +26,23 @@ export {
+export {
+  default as MyHistoryUpcomingTrips
+}
+  from '@views/user/history-upcoming-trips/history-upcoming-trips.view.vue';
```
do we realy need separate view for futute and past trips lists?

horba commented on front/src/views/index.js:
```diff
@@ -26,3 +26,23 @@ export {
+export {
+  default as MyHistoryTripRoute
+}
+  from '@views/user/history-trip-route/history-trip-route.view.vue';
```
it is just trip route

horba commented on front/src/views/index.js:
```diff
@@ -26,3 +26,23 @@ export {
+export {
+  default as MyHistoryAddTripRoute
+}
+  from '@views/user/history-trip-add-route/history-trip-add-route.view.vue';
```
I beleave that we don't need for creating new route separate view - is is usual route with empty state, isn't it?

horba replied:
no history of name - it is just details of routing and it is unnecessary info in name of view

valerachuk replied:
no, it's waypoint create/edit form

horba commented on front/src/views/index.js:
```diff
@@ -26,3 +26,23 @@ export {
+export {
+  default as MyHistoryAddTrip
+}
+  from '@views/user/history-add-trip/history-add-trip.view.vue';
```
no history in name

horba commented on front/src/views/user/history-add-trip/history-add-trip.view.js:
```diff
@@ -0,0 +1,53 @@
+        currency: this.newTripValues.currency,
+        flightTime: this.newTripValues.flightTime,
+        transplantTime: this.newTripValues.transplantTime,
+        differenceInTime: this.newTripValues.differenceInTime
```
you can try to use destruction
https://medium.com/@stasonmars/%D0%B4%D0%B5%D1%81%D1%82%D1%80%D1%83%D0%BA%D1%82%D1%83%D1%80%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D1%8F-%D0%B2-es6-%D0%BF%D0%BE%D0%BB%D0%BD%D0%BE%D0%B5-%D1%80%D1%83%D0%BA%D0%BE%D0%B2%D0%BE%D0%B4%D1%81%D1%82%D0%B2%D0%BE-b865bb71f376
https://learn.javascript.ru/destructuring-assignment

horba commented on server/Entities/WaypointFileRepository.cs:
```diff
@@ -0,0 +1,42 @@
+{
+  public class WaypointFileRepository
+  {
+    private readonly RepositoryContext _repositoryContext;
```
IRepositoryContext 
and use DI

horba commented on server/Entities/WaypointRepository.cs:
```diff
@@ -0,0 +1,56 @@
+{
+  public class WaypointRepository
+  {
+    private readonly RepositoryContext _repositoryContext;
```
IRepositoryContext

horba commented on server/WebAPI/Controllers/WaypointFileController.cs:
```diff
@@ -0,0 +1,61 @@
+  {
+
+    private readonly WaypointFileService _waypointFileService;
+    private readonly WaypointService _waypointService;
```
IWaypointFileService 
IWaypointService

valerachuk committed:
add waypoint mock data for prev trip

add file-attachment to add-waypoint form
rework add-file form design
rework form-validation logic
remove space-between from trip-list

valerachuk committed:
#71 fix waypoint close button icon

no "wp" shortcuts more
merge prev & next trip lists
rename trip & waypoint forms
use destruction in trip & waypoint forms
DI in RepositoryContext, WaypointServices, WaypointRepositories

valerachuk committed:
#71 waypoint date localization

valerachuk committed:
#71 delete time service :confused:

valerachuk committed:
split waypoint card into waypoint info & transfer info

valerachuk committed:
#71 delete custom date, time pickers

valerachuk committed:
#71 delete getter from waypoint store & move logic to ui

valerachuk committed:
#71 delete checkbox logic from store

sanioooook commented on front/src/components/mmt-waypoint-transfer-card/mmt-waypoint-transfer-card.component.html:
```diff
@@ -0,0 +1,15 @@
+<div class="path-info">
+  <div class="vertical-line"></div>
+  <v-icon class="transport-icon" :class="{'rotate-180': transfer.transport === 0}">{{iconMap[transfer.transport]}}
+  </v-icon>
```
you can specify a color for the icon. I don't really understand why to do this with a CSS variable if we already have this color defined in vuetify. I suggest just putting <v-icon color= "success base"/> and removing the CSS variable.

sanioooook commented on front/src/components/mmt-waypoint-transfer-card/mmt-waypoint-transfer-card.component.css:
```diff
@@ -0,0 +1,35 @@
+.path-info .vertical-line {
+  width: 0;
+  height: 36px;
+  border: 1px solid #6c6c6c;
```
but this color could be put in CSS variables

sanioooook commented on front/src/views/user/add-edit-route-form/add-edit-route-form.view.css:
```diff
@@ -0,0 +1,68 @@
+.v-form {
+  width: 248px;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
```
We have https://vuetifyjs.com/ru/styles/flex/
why you don't use it?

valerachuk committed:
#71 add image preview for image-files in waypoint card

valerachuk committed:
#71 fix waypoint - pathTime field

sanioooook commented on server/Entities/RepositoryContext.cs:
```diff
@@ -2,81 +2,97 @@
 namespace Entities
 {
-    public class RepositoryContext : DbContext
+  public interface IRepositoryContext
+  {
+    DbSet<User> Users { get; set; }
+    DbSet<Ticket> Tickets { get; set; }
+    DbSet<TicketRoute> TicketRoutes { get; set; }
+    DbSet<Country> Countries { get; set; }
+    DbSet<Language> Languages { get; set; }
+    DbSet<Trip> Trips { get; set; }
+    DbSet<Waypoint> Waypoints { get; set; }
+    DbSet<WaypointFile> WaypointFiles { get; set; }
+
+    int SaveChanges();
+  }
```
I think I have heard somewhere that each class is a separate file, each interface is a separate file, regardless of where it is implemented

sanioooook commented on server/WebAPI/Controllers/TripsController.cs:
```diff
@@ -19,6 +21,14 @@ public TripsController(TripService tripService)
       _tripService = tripService;
     }
 
+    [Authorize]
```
you can put it at the beginning of the controller if all methods need authorization

sanioooook commented on server/WebAPI/DTO/Trip/TripRequestDTO.cs:
```diff
@@ -0,0 +1,21 @@
+
+    public string FlightTime { get; set; }
+    public string TransplantTime { get; set; }
+    public string DifferenceInTime { get; set; }
```
it would be good to add attributes https://docs.microsoft.com/ru-ru/dotnet/api/system.componentmodel.dataannotations?view=netframework-4.8

valerachuk replied:
You mean validation

sanioooook replied:
yes, validation attributes

sanioooook commented on server/WebAPI/Controllers/WaypointFileController.cs:
```diff
@@ -0,0 +1,86 @@
+    [HttpPost]
+    [Route("multiple/{id}")]
+    [Consumes("multipart/form-data")]
+    public async Task<IActionResult> UploadFiles(int id, [FromForm] IFormFileCollection files)
```
why use this id if the method doesn't respond to it? you need to add [FromRoute] or remove it from the router

valerachuk replied:
It works without explicit indication. If we want to add explicit argument indication, we have to add it into all controllers, even [FromBody]

horba commented on front/src/store/modules/waypoints.store.js:
```diff
@@ -0,0 +1,94 @@
+          headers: {
+            'Content-Type': 'multipart/form-data'
+          }
+        });
```
add method to api service to upload files

horba commented on front/src/views/user/add-edit-route-form/add-edit-route-form.view.js:
```diff
@@ -0,0 +1,149 @@
+                nextWaypoint
+                 = waypoints[waypoints.findIndex(waypoint => waypoint === currentWaypoint) + 1];
+
+          this.newWaypointValues.departureCity = currentWaypoint.city;
+          this.newWaypointValues.departureDate
+          = format(new Date(currentWaypoint.departureDate), 'yyyy-MM-dd');
+          this.newWaypointValues.departureTime
+          = format(new Date(currentWaypoint.departureDate), 'HH:mm');
+
+          this.newWaypointValues.arrivalCity = nextWaypoint.city;
+          this.newWaypointValues.arrivalDate
+          = format(new Date(currentWaypoint.arrivalDate), 'yyyy-MM-dd');
+          this.newWaypointValues.arrivalTime
+          = format(new Date(currentWaypoint.arrivalDate), 'HH:mm');
+
+          this.newWaypointValues.pathLength = currentWaypoint.pathLength;
+          this.newWaypointValues.pathTime
+          = `${currentWaypoint.pathTime.hours}:${currentWaypoint.pathTime.minutes}`;
+          this.newWaypointValues.details = currentWaypoint.details;
+          this.newWaypointValues.transport = currentWaypoint.transport;
```
do we need it yet?

valerachuk replied:
yes, because date(time)Pickers works with strings, formatted like this: "yyyy-MM-dd'" ("HH:mm")

horba commented on server/WebAPI/Services/WaypointService.cs:
```diff
@@ -0,0 +1,189 @@
+
+    private readonly IWaypointRepository _waypointRepository;
+    private readonly IWaypointFileService _waypointFileService;
+    private readonly GooglePlacePhotoService _photoService;
```
IGooglePlacePhotoService?

horba commented on server/WebAPI/Services/WaypointService.cs:
```diff
@@ -0,0 +1,189 @@
+        {
+          _waypointRepository.DeleteWaypoint(_waypointRepository.GetWaypoints().Where(wp => wp.TripId == wpDTO.TripId).OrderByDescending(wp => wp.Order).FirstOrDefault());
+          wpCount--;
+        }
```
too long line - place each linq statement at separate line

horba commented on server/WebAPI/Services/WaypointService.cs:
```diff
@@ -0,0 +1,189 @@
+    }
+
+  }
+}
```
please, no wp

valerachuk committed:
#71 add validation to trip & waypoint

add postFile method to service
modify time validation rules
DI for IGooglePlacePhotoService
delete "wp" shortcuts

horba commented on server/WebAPI/Controllers/WaypointsController.cs:
```diff
@@ -0,0 +1,104 @@
+  {
+
+    private readonly IWaypointService _waypointService;
+    private readonly TripService _tripService;
```
ITripService

horba commented on server/WebAPI/Services/TripService.cs:
```diff
@@ -4,18 +4,25 @@
   public class TripService
   {
     private readonly IMapper _mapper;
     private readonly TripRepository _tripRepository;
```
ITripRepository - for all dependencies - use DI

valerachuk committed:
#71 add DI to TripService & TripRepository

valerachuk committed:
#71 add DI to GoogleOauthService
